### DISCUSSION TITLE
Connecting KOReader to Anki

### DISCUSSION THREAD

**Discussion by ichnilatis-gr** [koreader/koreader#12538] [Q&A]
Dear @Steven630 could you explain to me how I could connect KOReader to Anki? I have installed AnkiConnect into Anki and I have installed Anki plug-in into KOReader. But I don't know how to connect these two. I can't find Anki settings in KOReader.
Thank you!

**Comment by Steven630** [comment]
Hi, are you using this plugin?

https://github.com/Ajatt-Tools/anki.koplugin

**Reply by ichnilatis-gr** [reply]
Yes.

**Reply by Steven630** [reply]
Set AnkiConnect ip address to  0.0.0.0. In the plugin's config.lua, set the ip address to your desktop's ipv4 address (If you are on Windows, use cmd—ipconfig to see your ip).

**Reply by ichnilatis-gr** [reply]
Could you give me a screenshot of the configuration in config.lua as an example, please? I can't find where I have to write the ip address.
Otherwise, I'm afraid I have to give up. The whole process is somewhat incomprehensible to me.
Thanks!

**Reply by Steven630** [reply]
> Could you give me a screenshot of the configuration in config.lua as an example, please? I can't find where I have to write the ip address.
> Otherwise, I'm afraid I have to give up. The whole process is somewhat incomprehensible to me.
> Thanks!

Of course. The configuration file is under the plugin's folder.  In the second screenshot you should change the ip address according to your desktop's, not 0.0.0.0 (should be something like 192.168...).  At first I set it to all zero and got a connection refused error. This issue might also help: https://github.com/Ajatt-Tools/anki.koplugin/issues/29

Here are two screenshots:

![Screenshot_20240921_212313.jpg](https://github.com/user-attachments/assets/d19cd8a6-8ad1-4512-97a6-3edf3bd9cb93)

![Screenshot_20240921_212354.jpg](https://github.com/user-attachments/assets/1d05e6bd-3d9c-417e-8cbd-be5e2067f003)


Make sure to fill in all the required fields. If you still have trouble, you can share your screenshot.

**Reply by ichnilatis-gr** [reply]
Dear @Steven630, After a long time I tried again to activate the anki plugin, after installing the latest version and making the necessary settings. However when I try to sync with AnkiConnect the following message comes up. What do you think is the problem? Which settings should I do exactly in AnkiConnect?
Thank you in advance for your help and patience.

![scr0002](https://github.com/user-attachments/assets/3496145f-7162-447d-b20e-6fcdcfba30be)

**Reply by lionrayonnant** [reply]
Hey, can u please give some screenshots of your modifications ?

**Reply by ichnilatis-gr** [reply]
The settings of the profile are:

```
-- This file contains all the user configurable options
-- Entries which aren't marked as REQUIRED can be ommitted completely
local Config = {
    ----------------------------------------------
    ---- [[ GENERAL CONFIGURATION OPTIONS ]] -----
    ----------------------------------------------
    -- This refers to the IP address of the PC ankiconnect is running on
    -- Remember to expose the port ankiconnect listens on so we can connect to it
    -- [REQUIRED] The ankiconnect settings also need to be updated to not only listen on the loopback address
    url = "http://192.168.x.xxx:8765",
    -- [REQUIRED] name of the anki deck
    deckName = "Αγγλικά",
    -- [REQUIRED] note type of the notes that should be created
    modelName = "Αγγλικά (κατ' επιλογή αντεστραμένη κάρτα)",
    -- Each note created by the plugin will have the tag 'KOReader', it is possible to add other custom tags
    -- A card with custom tags can be created by pressing and holding the 'Add to Anki' button, which pops up a menu with some extra options.
    custom_tags = { "NEEDS_WORK" },

    -- It is possible to toggle whether duplicate notes can be created. This can be of use if your note type contains the full sentence as first field (meaning this gets looked at for uniqueness)
    -- When multiple unknown words are present, it won't be possible to add both in this case, because the sentence would be the same.
    allow_dupes = true,
    -- The scope where ankiconnect will look to to find duplicates
    dupe_scope = "deck",
    -- api key - extra authentication supported by ankiconnect, see https://git.foosoft.net/alex/anki-connect#authentication
    -- this is totally optional and probably unnecessary, unless you expose anki-connect on the public network for some reason
    api_key = nil,


    ----------------------------------------------
    --- [[ NOTE FIELD CONFIGURATION OPTIONS ]] ---
    ----------------------------------------------
    -- [REQUIRED] The field name where the word which was looked up in a dictionary will be sent to.
    word_field = "Αγγλικά",

    -- The field name where the sentence in which the word we looked up occurred will be sent to.
    context_field = "Αγγλικά",

    -- Translation of the context field
    translated_context_field = "Ελληνικά",

    -- Amount of sentences which are prepended to the word looked up. Set this to 1 to complete the current sentence.
    prev_sentence_count = 1,

    -- Amount of sentences which are appended to the word looked up. Set this to 1 to complete the current sentence.
    next_sentence_count = 1,

    -- [REQUIRED] The field name where the dictionary definition will be sent to.
    def_field = "Συμπληρωματικά",

    -- The field name where metadata (book source, page number, ...) will be sent to.
    -- This metadata is parsed from the EPUB's metadata, or from the filename
    meta_field = "Συμπληρωματικά",

    -- The plugin can query Forvo for audio of the word you just looked up.
    -- The field name where the audio will be sent to.
    audio_field = "Προφορά",

    -- list of extensions which should be enabled, by default they are all off
    -- an extension is turned on by listing its filename in the table below
    -- existing extensions are listed below, remove the leading -- to enable them
    enabled_extensions = {
        --"EXT_dict_edit.lua",
        --"EXT_dict_word_lookup.lua",
        --"EXT_multi_def.lua",
        --"EXT_pitch_accent.lua"
    }
}
return Config
```
And the settings of AnkiConnect are: 

```
{
    "apiKey": null,
    "apiLogPath": null,
    "ignoreOriginList": [],
    "webBindAddress": "0.0.0.0",
    "webBindPort": 8765,
    "webCorsOriginList": [
        "http://localhost"
    ]
}
```

I am not sure what I have to write in the place of "localhost". I put the IPv4, but it did'n't work.
Thank you for your willingness to help.

**Reply by ichnilatis-gr** [reply]
Any idea?

**Reply by Frenzie** [reply]
The file you posted suggests the IPv4 of the computer that's running on should work. The ports are all default?

**Reply by ichnilatis-gr** [reply]
Yes, it's 8765.

**Reply by Steven630** [reply]
You may have to ask the author of the plug-in. I haven't used it for the past months.

**Comment by shuvashish76** [comment]
@ichnilatis-gr Just tried latest AnkiConnect on both Kobo and Desktop both works as expected.

[Anki+KOReader.webm](https://github.com/user-attachments/assets/71597cb3-9476-44ec-8fc8-29aa0de98955)

**Reply by ichnilatis-gr** [reply]
My question is whether I have made the correct settings, as they are given above. AnkiConnect is not working on my device and I can't figure out what the problem is. I keep getting the "Synchronizing failed!" message.

**Reply by shuvashish76** [reply]
Yes your AnkiConnect configuration at desktop is correct. But nobody knows if your KOReader Anki plugin **profile** configuration match with Desktop one as you didn't share your desktop configuration.

- I'd recommend to fresh install the plugin in KOReader as it has some new changes, also remove the user patch it's not necessary anymore. Now everything can be done from GUI no need to manually create profile `.lua` file.
- 1st from GUI "anki-connect settings" option use the correct IPv4 URL (REQUIRED) and click "Test", it should show "Connection successful!"...if not then you're still putting the IP:Port wrong.
- I think in setting up Profile step you're doing it wrong follow me for simplicity (just to get an idea):
  - From desktop Anki create your **Deck** (REQUIRED) e.g. `Vocabs`, click Add choose **Type** (REQUIRED) to `Basic`, click **Fields** (REQUIRED) add following four `Word`, `Context`, `Metadata`, `Glossary`. Fill those fields, add your 1st random Card for test.
  - Now from KOReader anki plugin GUI (available under Settings > Anki Settings > Edit profiles > Create new profile)
    - General Settings
      - **Anki Deckname** : `Vocabs`, **Anki Note Type** : `Basic`
    - Anki Note Settings
      - **Word Field** : `Word`, **Context Field** : `Context`, **Glossary Field** : `Glossary`, **Metadata Field** : `Metadata`

Finally open any book > long click any word > From Dictionary window "Add to Anki".
Let me know if it works ;)

**Reply by ichnilatis-gr** [reply]
Unfortunately, I can't seem to get it to work. I'll probably have to give up. That's OK. Thanks for your help anyway!

**Reply by shuvashish76** [reply]
Random guess. If you're using Firewall you've to whitelist that port. On my [Firewalld](http://www.firewalld.org/) I've listed the 8765 TCP port.
